.github/workflows/prepend-license.yml

```diff
@@ -0,0 +1,36 @@
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+name: Prepend licences to files
+jobs:
+  prepend_license: 
+    name: Commit prepended licences to files 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.GH_PR_TOKEN }}
```

Comment:
TODO: need to get a token

also, for anyone picking this up, they can test it directly on this branch

Reply:
done